packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -5882,7 +5901,57 @@ export default class Meeting extends StatelessWebexPlugin {
       throw error;
     }
 
-    await this.waitForMediaConnectionConnected();
+    try {
+      await this.waitForMediaConnectionConnected();
+
+      if (this.retriedWithTurnServer) {
+        Metrics.sendBehavioralMetric(
+          BEHAVIORAL_METRICS.RETRY_MEDIA_CONNECTION_WITH_TURN_SERVER_SUCCESS,
+          {
+            correlation_id: this.correlationId,
+          }
+        );
+      }
+    } catch (error) {
+      // If waitForMediaConnectionConnected() fails when we haven't done turn discovery (and the config allows for turn discovery to happen) then we
+      // attempt to establish a media connection again but this time using turn discovery. If we used turn discovery on the first pass we do not attempt
+      // connection again.
+
+      // @ts-ignore - config coming from registerPlugin
+      if (!this.turnServerUsed && this.turnDiscoverySkippedReason !== 'config') {
```

Comment:
can the code in this function (just the catch block) be split up further ? i'm thinking a catch handler function which calls out to it's steps. 

This is a nitpick and just intended for cleanliness, not a blocker

Reply:
Done, moved this logic out to ```handleWaitForMediaConnectionConnectedError```

Reply:
Broke this up more as per 1-1 for cleanliness